homeassistant/components/mqtt/event.py

```diff
@@ -133,6 +137,7 @@ def _prepare_subscribe_topics(self) -> None:
 
         @callback
         @log_messages(self.hass, self.entity_id)
+        @write_state_on_attr_change(self, {"state"})
```

Comment:
Had to look though this one a bit, but it does look like the `state` has the time in it so it will always change so tracking it is enough

Reply:
That is correct, that is why I renamed the title.
It is more meant to have one pattern for all mqtt message handlers.